app/invocation/invocation_action_card.tsx

```diff
@@ -90,103 +93,65 @@ export default class InvocationActionCardComponent extends React.Component<Props
   }
 
   fetchInputRoot(rootDigest: build.bazel.remote.execution.v2.IDigest) {
-    let inputRootFile =
-      "bytestream://" +
-      this.props.model.getCacheAddress() +
-      "/blobs/" +
-      this.props.model.getDigestFunctionDir() +
-      rootDigest.hash +
-      "/" +
-      rootDigest.sizeBytes;
+    let inputRootURL = this.props.model.getBytestreamURL(rootDigest);
     rpcService
-      .fetchBytestreamFile(inputRootFile, this.props.model.getInvocationId(), "arraybuffer")
+      .fetchBytestreamFile(inputRootURL, this.props.model.getInvocationId(), "arraybuffer")
       .then((buffer: any) => {
-        let tempRoot = build.bazel.remote.execution.v2.Directory.decode(new Uint8Array(buffer));
-        let inputDirs: InputNode[] = tempRoot.directories.map(
-          (node) =>
-            ({
-              obj: node,
-              type: "dir",
-            } as InputNode)
-        );
-        this.setState({
-          inputRoot: tempRoot,
-          inputDirs: inputDirs,
-        });
+        let inputRoot = build.bazel.remote.execution.v2.Directory.decode(new Uint8Array(buffer));
+        let inputDirs: InputNode[] = inputRoot.directories.map((node) => ({
+          obj: node,
+          type: "dir",
+        }));
+        this.setState({ inputRoot, inputDirs });
       })
       .catch((e) => console.error("Failed to fetch input root:", e));
   }
 
   fetchActionResult() {
-    let digestParam = this.props.search.get("actionResultDigest");
-    if (digestParam == null) {
-      digestParam = this.props.search.get("actionDigest");
+    let digestParam = this.props.search.get("actionResultDigest") ?? this.props.search.get("actionDigest");
```

Comment:
Should be fine, but just a reminder that this would short-circuit for `""` instead of checking `actionDigest`

Reply:
ack, I think it's fine as-is since it's probably a bug if the URL contains `actionDigest` set to an empty string